Libraries/src/Amazon.Lambda.Annotations/README.md

```diff
@@ -0,0 +1,149 @@
+{
+    [LambdaFunction]
+    [RestApi("/plus/{x}/{y}")]
+    public int Plus(int x, int y)
+    {
+        return x + y;
+    }
+}
+```
+
+## Source Generator
+
+To bridge the gap from Lambda Annotations programming model to the normal programming model a .NET source generator is included in this package.
+After adding the attributes to your .NET code the source generator will generate the translation code between the 2 programming models. It will also
+keep in sync the generated information including a new function handler string into the CloudFormation template. The usage of source
+generator is transparent to the user.
+
+To see the code that is generated by the source generator turn the verbosity to detailed when executing a build. From the command this 
+is done by using the `--verbosity` switch.
+```
+dotnet build --verbosity detailed
+```
+To change the verbosity in Visual Studio go to Tools -> Options -> Projects and Solutions and adjust the MSBuild verbosity drop down boxes.
+
+## Dependency Injection
+
+Lambda Annotations supports using dependency injection. A class can be marked with a `LambdaStartup` attribute. The class will 
+have a `ConfigureServices` method for configuring services.
+
+The services can be injected by either constructor injection or using the `FromServices` attribute on a method parameter of
+the function decorated with the `LambdaFunction` attribute.
+
+Services injected via the constructor have a lifecycle for the length of the Lambda compute container. For each Lambda 
+invocation a scope is created and the services injected using the `FromServices` attribute are created within the scope.
+
+Example startup class:
+```csharp
+[LambdaStartup]
+public class Startup
+{
+    public void ConfigureServices(IServiceCollection services)
+    {
+        services.AddAWSService<Amazon.S3.IAmazonS3>();
+        services.AddScoped<ITracker, DefaultTracker>();
+    }
+}
+```
+
+Example function using DI:
+```csharp
+public class Functions
+{
+    IAmazonS3 _s3Client;
+
+    public Functions(IAmazonS3 s3Client)
+    {
+        _s3Client = s3Client;
+    }
+
+
+    [LambdaFunction]
+    [HttpApi(HttpMethod.Put, HttpApiVersion.V2, "/process/{name}")]
+    public async Task Process([FromServices] ITracker tracker, string name, [FromBody] string data)
+    {
+        tracker.Record();
+
+        await _s3Client.PutObjectAsync(new PutObjectRequest
+        {
+            BucketName = "storage-bucket",
+            Key = name,
+            ContentBody = data
+        });
+    }
+}
+```
+
+## Lambda .NET Attributes
+
+List of .NET attributes currently supported.
+
+
+* LambdaFunction
+    * Placed on a method. Indicates this method should be exposed as a Lambda function.
+* LambdaStartup
+    * Placed on a class. Indicates this type should be used as the startup class and is used to configure the dependency injection and middleware. There can only be one class in a Lambda project with this attribute.
+
+### Event Attributes    
```

Comment:
Should include one liner for no event attached lambda function. They can also be written using Annotations and take the advantage of template generation, DI etc.

Reply:
Add a description to the event attribute section and mentioned that you don't have to have an event attribute.